tests/test_hamiltonian_data.py

```diff
@@ -447,19 +450,29 @@ def test_noisy_interaction_matrix():
         ),
         "ch0",
     )
-    noise = pulser.NoiseModel(state_prep_error=1.0, runs=1)
+    noise = pulser.NoiseModel(state_prep_error=0.5, runs=1)
     ham = HamiltonianData.from_sequence(seq, noise_model=noise)
     assert np.allclose(
         ham.noisy_interaction_matrix._array,
-        np.zeros_like(ham._interaction_matrix),
+        np.array(
+            [
+                [0.0000, 0.0000, 0.0000, 0.0000],
+                [0.0000, 0.0000, 26.4198, 0.0000],
+                [0.0000, 26.4198, 0.0000, 0.0000],
+                [0.0000, 0.0000, 0.0000, 0.0000],
+            ]
+        ),
```

Comment:
nit: Asserting that the bad atoms and the first and forth and maybe comparing with the (noiseless) interaction matrix would be good additions imo